lib/Controller/ConsentAdmin.php

```diff
@@ -0,0 +1,418 @@
+        $this->metadataStorageHandler = MetaDataStorageHandler::getMetadataHandler();
+        $this->session = $session;
+    }
+
+
+    /**
+     * Inject the \SimpleSAML\Module\consent\Store dependency.
+     *
+     * @param \SimpleSAML\Module\consent\Store $store
+     */
+    public function setStore(Store $store): void
+    {
+        $this->store = $store;
+    }
+
+
+    /**
+     * Inject the \SimpleSAML\Auth\Simple dependency.
+     *
+     * @param \SimpleSAML\Auth\Simple $authSimple
+     */
+    public function setAuthSimple(Auth\Simple $authSimple): void
+    {
+        $this->authSimple = $authSimple;
+    }
+
+
+    /**
+     * Inject the \SimpleSAML\Metadata\MetaDataStorageHandler dependency.
+     *
+     * @param \SimpleSAML\Metadata\MetaDataStorageHandler $handler
+     */
+    public function setMetadataStorageHandler(MetadataStorageHandler $handler): void
+    {
+        $this->metadataStorageHandler = $handler;
+    }
+
+
+    /**
+     * Inject the \SimpleSAML\Module\consent\Auth\Process\Consent dependency.
+     *
+     * @param \SimpleSAML\Module\consent\Auth\Process\Consent $consent
+     */
+    public function setConsent(Consent $consent): void
+    {
+        $this->consent = $consent;
+    }
+
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request The current request.
+     *
+     * @return \SimpleSAML\XHTML\Template
+     */
+    public function main(Request $request): Template
+    {
+        $authority = $this->moduleConfig->getValue('authority');
+
+        $as = new $this->authSimple($authority);
+
+        // If request is a logout request
+        $logout = $request->get('logout');
+        if ($logout !== null) {
+            $returnURL = $this->moduleConfig->getValue('returnURL');
+            $as->logout($returnURL);
+        }
+
+        $hashAttributes = $this->moduleConfig->getValue('attributes.hash', false);
+
+        $excludeAttributes = $this->moduleConfig->getValue('attributes.exclude', []);
+
+        // Check if valid local session exists
+        $as->requireAuth();
+
+        // Get released attributes
+        $attributes = $as->getAttributes();
+
+        // Get metadata storage handler
+        $metadata = $this->metadataStorageHandler;
+
+        /*
+         * Get IdP id and metadata
+         */
+        $idp_entityid = $metadata->getMetaDataCurrentEntityID('saml20-idp-hosted');
+        $idp_metadata = $metadata->getMetaData($idp_entityid, 'saml20-idp-hosted');
+
+        // Calc correct source
+        if ($as->getAuthData('saml:sp:IdP') !== null) {
+            // from a remote idp (as bridge)
+            $source = 'saml20-idp-remote|' . $as->getAuthData('saml:sp:IdP');
+        } else {
+            // from the local idp
+            $source = $idp_metadata['metadata-set'] . '|' . $idp_entityid;
+        }
+
+        // Get user ID
+        if (isset($idp_metadata['userid.attribute']) && is_string($idp_metadata['userid.attribute'])) {
```

Comment:
Is this still how we call it?

Reply:
Good catch!  This one is a goner and every authproc should have it's own 'identifyingAttribute'..  In this case it should probably inherit this setting from the consent-module